src/librustc_back/target/mod.rs

```diff
@@ -580,6 +585,20 @@ impl Target {
                 Some(Ok(()))
             })).unwrap_or(Ok(()))
             } );
+            ($key_name:ident, RelroLevel) => ( {
+                let name = (stringify!($key_name)).replace("_", "-");
+                obj.find(&name[..]).and_then(|o| o.as_string().and_then(|s| {
+                    match s {
+                        "full" => base.options.$key_name = RelroLevel::Full,
+                        "partial" => base.options.$key_name = RelroLevel::Partial,
+                        "off" => base.options.$key_name = RelroLevel::Off,
```

Comment:
I was just duplicating the way it was done for `PanicStrategy`, but I can look into deduplicating it later this week.